feat: add HPA behaviour to troubleshooting guide (#662)
rancher/fleet: docs/troubleshooting.md

```json
{"op":"remove", "path":"/spec/template/spec/hostNetwork"}
```

> ### My bundle which has a Horizontal Pod Autoscaler is in state `Modified`

Reviewer: ```suggestion
### Bundle has a Horizontal Pod Autoscaler (HPA) in modified state
```

> For bundles with an HPA is expected that it is in state `Modified`, as the bundle contains fields that differ from the state of the Bundle at deployment.

Reviewer: ```suggestion
For bundles with an HPA, the expected state is `Modified`, as the bundle contains fields that differ from the state of the Bundle at deployment.
```

> In such cases you have to define a patch in the `fleet.yaml` to ignore this field according to [bundle-diffs](./bundle-diffs.md).

Reviewer: ```suggestion
You must define a patch in the `fleet.yaml` to ignore this field according to [bundle-diffs](./bundle-diffs.md).
```

Reviewer: Please specify what field you're referring to when you say `ignore this field`. Ty!

> An example of configuring this for the replicas field of a Deployment can be found [here](https://github.com/axeal/fleet-test/blob/320229/fleet.yaml).

Reviewer: ```suggestion
Please refer [here](https://github.com/axeal/fleet-test/blob/320229/fleet.yaml) for an example of how to configure settings for the **replicas** field of a Deployment.
```

Author: Thanks for the review and the corrections. Please check my latest changes.

1. You can also force update the `gitrepo` to perform a manual resync. Select **GitRepo** on the left navigation bar, then select **Force Update**.

### Bundle has a Horizontal Pod Autoscaler (HPA) in modified state

For bundles with an HPA, the expected state is `Modified`, as the bundle contains fields that differ from the state of the Bundle at deployment - usually `ReplicaSet`.

You must define a patch in the `fleet.yaml` to ignore this field according to [`GitRepo` or `Bundle` stuck in modified state](#gitrepo-or-bundle-stuck-in-modified-state).

> Here an example of such a patch for the deployment `nginx` in namespace `default`

Reviewer: ```suggestion
Here is an example of such a patch for the deployment `nginx` in namespace `default`:
```

Reviewer: Thanks! Just a minor correction on this line.
